pkg/modulewriter/modulewriter.go

```diff
@@ -115,17 +119,25 @@ func WriteDeployment(blueprint *config.Blueprint, outputDir string, overwriteFla
 	return nil
 }
 
+func createGroupDirs(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) error {
+	deploymentio := deploymentio.GetDeploymentioLocal()
+	for _, grp := range *deploymentGroups {
+		groupPath := filepath.Join(deploymentPath, grp.Name)
+		// Create the deployment group directory if not already created.
+		if _, err := os.Stat(groupPath); errors.Is(err, os.ErrNotExist) {
+			if err := deploymentio.CreateDirectory(groupPath); err != nil {
+				return fmt.Errorf("failed to create directory at %s for deployment group %s: err=%w",
```

Comment:
After taking a closer look at all of this, I feel like there is a smell here. We are:
- checking the path does not exist (here) and skipping if it does
- checking the path does not exist (deploymentio.CreateDirectory) and erroring if it does
- os.MkdirAll already handles case that dir already exists [ref](https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/os/path.go;l=16-17;drc=2291cae2af659876e93a3e1f95c708abb1475d02).

Personally I don't really understand the need for deploymentio to manage CreateDirectory. Without it, this would become the following:

```

L125	groupPath := filepath.Join(deploymentPath, grp.Name)
	if err := os.MkdirAll(groupPath, 0755); err != nil {
		return fmt.Errorf("failed to create directory at %s for deployment group %s: err=%w", 
			groupPath, grp.Name, err)}
```

Reply:
I agree, partially. blueprintio wasn't needed and obfuscated some of the logic needlessly. However, I would rather use Mkdir over MkdirAll. The expectation is that the deployment directory already exists, so if for some reason that fails to work in the future, I'd rather it fail than be covered up by using an overly permissive MkdirAll command here.